File: Programmers/cpp/LEVEL_2_가장_큰_수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

bool compare(string& a, string& b) {
	if (a + b <= b + a) {
		return false;
	}

	return true;
}

bool allZeroCheck(vector<int> numbers) {
	for (int n : numbers) {
		if (n != 0) {
			return false;
		}
	}

	return true;
}

string solution(vector<int> numbers) {
	string answer = "";

	if (allZeroCheck(numbers)) {
		return "0";
	}

	vector<string> v;

	for (int i = 0; i < numbers.size(); i++) {
		v.push_back(to_string(numbers[i]));
	}

	sort(v.begin(), v.end(), compare);

	for (string s : v) {
		answer += s;
	}

	return answer;
}

File: Programmers/java/LEVEL_2_가장_큰_수.java
import java.util.*;

class Solution {
    private boolean allZeroCheck(int[] numbers) {
        for (int n : numbers) {
            if (n != 0) {
                return false;
            }
        }

        return true;
    }

    public String solution(int[] numbers) {
        String answer = "";

        if (allZeroCheck(numbers)) {
            return "0";
        }

        List<String> list = new ArrayList<>();

        for (int i = 0; i < numbers.length; i++) {
            list.add(String.valueOf(numbers[i]));
        }

        Collections.sort(list, (x, y) -> {
            if ((x + y).compareTo(y + x) < 0) {
                return 1;
            }

            if ((x + y).compareTo(y + x) > 0) {
                return -1;
            }

            return 0;
        });

        for (String s : list) {
            answer += s;
        }

        return answer;
    }
}